scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/ReporterOps.scala

```diff
@@ -1,11 +1,14 @@
 package scala.meta.internal
 package semanticdb
 
+import scala.collection.mutable
 import scala.tools.nsc.reporters.StoreReporter
 import scala.reflect.internal.util.{Position => gPosition}
 
 trait ReporterOps { self: DatabaseOps =>
-
+  // Hack, keep track of how many messages we have returns for each path to avoid
+  // duplicate messages. The key is System.identityHashCode to keep memory usage low.
+  private val returnedMessagesByPath = mutable.LongMap.empty[Int]
```

Comment:
I think we can use `unit` as a key. Units stay in memory for the entire duration of the compilation pipeline.

Reply:
Done.